src/solvers/dgsem_tree/containers_3d.jl

```diff
@@ -812,4 +812,217 @@ function init_mortars!(mortars, elements, mesh::TreeMesh3D)
+                                               (n_variables, n_nodes + 1, n_nodes,
+                                                n_nodes,
+                                                capacity))
+    resize!(_antidiffusive_flux1_R,
+            n_variables * (n_nodes + 1) * n_nodes * n_nodes * capacity)
+    fluxes.antidiffusive_flux1_R = unsafe_wrap(Array, pointer(_antidiffusive_flux1_R),
+                                               (n_variables, n_nodes + 1, n_nodes,
+                                                n_nodes,
+                                                capacity))
+    resize!(_antidiffusive_flux2_L,
+            n_variables * n_nodes * (n_nodes + 1) * n_nodes * capacity)
+    fluxes.antidiffusive_flux2_L = unsafe_wrap(Array, pointer(_antidiffusive_flux2_L),
+                                               (n_variables, n_nodes, n_nodes + 1,
+                                                n_nodes,
+                                                capacity))
+    resize!(_antidiffusive_flux2_R,
+            n_variables * n_nodes * (n_nodes + 1) * n_nodes * capacity)
+    fluxes.antidiffusive_flux2_R = unsafe_wrap(Array, pointer(_antidiffusive_flux2_R),
+                                               (n_variables, n_nodes, n_nodes + 1,
+                                                n_nodes,
+                                                capacity))
+
+    resize!(_antidiffusive_flux3_L,
+            n_variables * n_nodes * n_nodes * (n_nodes + 1) * capacity)
+    fluxes.antidiffusive_flux3_L = unsafe_wrap(Array, pointer(_antidiffusive_flux3_L),
+                                               (n_variables, n_nodes, n_nodes,
+                                                n_nodes + 1,
+                                                capacity))
+    resize!(_antidiffusive_flux3_R,
+            n_variables * n_nodes * n_nodes * (n_nodes + 1) * capacity)
+    fluxes.antidiffusive_flux3_R = unsafe_wrap(Array, pointer(_antidiffusive_flux3_R),
+                                               (n_variables, n_nodes, n_nodes,
+                                                n_nodes + 1,
+                                                capacity))
+    return nothing
+end
+
+# TODO: Does it make sense to implement the alpha1/2/3 dimension independent
+# with indices like [**direction**, i, j, k, elements]?
+# Container data structure (structure-of-arrays style) for variables used for IDP limiting
+mutable struct ContainerSubcellLimiterIDP3D{uEltype <: Real}
+    alpha::Array{uEltype, 4} # [i, j, k, element]
+    alpha1::Array{uEltype, 4}
+    alpha2::Array{uEltype, 4}
+    alpha3::Array{uEltype, 4}
+    variable_bounds::Dict{Symbol, Array{uEltype, 4}}
+    # internal `resize!`able storage
+    _alpha::Vector{uEltype}
+    _alpha1::Vector{uEltype}
+    _alpha2::Vector{uEltype}
+    _alpha3::Vector{uEltype}
+    _variable_bounds::Dict{Symbol, Vector{uEltype}}
+end
```

Comment:
Does it make sense to combine `alpha1/2/3` into one variable? That would allow the same container for 2d and 3d.
One problem is that those alphas have different sizes
`alpha1` has size `(n_nodes + 1, n_nodes, n_nodes, n_elements)`
`alpha2` has size `(n_nodes, n_nodes + 1, n_nodes, n_elements)`
`alpha3` has size `(n_nodes, n_nodes, n_nodes + 1, n_elements)`

That could be combined to one big alpha of size `(n_dims, n_nodes + 1, n_nodes + 1, n_nodes + 1, n_elements)`

Reply:
On the other hand, the following parts would be not so clean anymore.
https://github.com/trixi-framework/Trixi.jl/blob/329d5f115c2e02114f3f5a4d768d909832656e82/src/solvers/dgsem_tree/subcell_limiters_2d.jl#L53-L66

https://github.com/trixi-framework/Trixi.jl/blob/329d5f115c2e02114f3f5a4d768d909832656e82/src/callbacks_stage/subcell_limiter_idp_correction_2d.jl#L22-L34

Reply:
Removed `alpha1/2/3` completely in https://github.com/trixi-framework/Trixi.jl/pull/2588